rdmo/management/management/commands/merge_attributes.py

```diff
@@ -0,0 +1,228 @@
+import logging
+from collections import namedtuple
+from textwrap import dedent
+
+from django.core.management.base import BaseCommand, CommandError
+from django.utils.translation import gettext_lazy as _
+
+from rdmo.conditions.models import Condition
+from rdmo.domain.models import Attribute
+from rdmo.management.utils import replace_uri_in_template_string
+from rdmo.projects.models import Value
+from rdmo.questions.models import Page, Question, QuestionSet
+from rdmo.tasks.models import Task
+from rdmo.views.models import View
+
+logger = logging.getLogger(__name__)
+
+
+ModelHelper = namedtuple('ModelHelper', ['model', 'lookup_field', 'field_introspection', 'replacement_function'])
+
+REPLACE_ATTRIBUTE_MODEL_HELPERS = [
+    ModelHelper(Page, 'attribute', False, None),
+    ModelHelper(QuestionSet, 'attribute', False, None),
+    ModelHelper(Question, 'attribute', False, None),
+    ModelHelper(Condition, 'source', False, None),
+    ModelHelper(Task, 'start_attribute', False, None),
+    ModelHelper(Task, 'end_attribute', False, None),
+    ModelHelper(Value, 'attribute', False, None),
+    ModelHelper(View, 'template', True, replace_uri_in_template_string)
+]
+
+
+def get_queryset(model, lookup_field, field_introspection, *_, attribute=None):
+    if attribute is None:
+        raise ValueError("attribute is required")
+    if field_introspection:
+        return model.objects.filter(**{f"{lookup_field}__contains": attribute.path})
+    return model.objects.filter(**{lookup_field: attribute})
+
+
+def replace_attribute_on_instance(instance, lookup_field, source, target, field_introspection, replacement_function):
+    instance_field = getattr(instance, lookup_field)
+
+    if isinstance(instance_field, Attribute) and instance_field != source:
+        raise ValueError("Instance attribute should be equal to the source attribute")
+    if not isinstance(instance_field, Attribute) and not field_introspection:
+        raise ValueError("When instance field is not an attribute, it should use field introspection")
+
+    if field_introspection:
+        replacement_field = replacement_function(instance_field, source, target)
+    else:
+        replacement_field = target
+
+    setattr(instance, lookup_field, replacement_field)
+    return instance
+
+
+def replace_attribute_on_element_instances(model,
```

Comment:
This function is also not needed, it is called only one time.

Reply:
yes, but now that you can read it from top to bottom, it is much better readable.